src/main/java/io/fabric8/maven/docker/util/AnsiLogger.java

```diff
@@ -48,17 +53,36 @@
     // Whether to use ANSI codes
     private boolean useAnsi;
 
-    public AnsiLogger(Log log, boolean useColor, boolean verbose) {
+    public enum LogVerboseCategory {
+        BUILD("build"), API("api");
+        private String category;
+
+        LogVerboseCategory(String category) {
+            this.category = category;
+        }
+
+        public String getValue() {
+            return category;
+        }
+    };
+
+    public AnsiLogger(Log log, boolean useColor, String verbose) {
         this(log, useColor, verbose, false);
     }
 
-    public AnsiLogger(Log log, boolean useColor, boolean verbose, boolean batchMode) {
+    public AnsiLogger(Log log, boolean useColor, String verbose, boolean batchMode) {
         this(log, useColor, verbose, batchMode, DEFAULT_LOG_PREFIX);
     }
 
-    public AnsiLogger(Log log, boolean useColor, boolean verbose, boolean batchMode, String prefix) {
+    public AnsiLogger(Log log, boolean useColor, String verbose, boolean batchMode, String prefix) {
         this.log = log;
-        this.verbose = verbose;
+        this.isVerbose = checkBackwardVersionValues(verbose); // For backward compatibility
```

Comment:
What happens when "verbose" is null ? It doesn't look like that the code really checks this case. It should be that in this case verbose logging should be disabled.